db/utils.py

```diff
@@ -1,9 +1,19 @@
+from db.records import exceptions
+from sqlalchemy.exc import ProgrammingError
+from psycopg2.errors import UndefinedFunction
+
+
 def execute_statement(engine, statement, connection_to_use=None):
-    if connection_to_use:
-        return connection_to_use.execute(statement)
-    else:
-        with engine.begin() as conn:
-            return conn.execute(statement)
+    try:
+        if connection_to_use:
+            return connection_to_use.execute(statement)
+        else:
+            with engine.begin() as conn:
+                return conn.execute(statement)
+    except ProgrammingError as e:
+        if isinstance(e.orig, UndefinedFunction):
```

Comment:
`execute_statement` is a generic function used for executing most of queries, wouldn't it affect other queries that throw the same `UndefinedFunction` exception and need more than the first line of the error message as this line assumes none of the `UndefinedFunction` need more than the first message? 

> It's not certain that the first line is the correct one for all relevant error messages

I was concerned about where we handle extracting the subset of the error, not the way we process it. There are various places where we can extract the underlying error, just want to make sure we have defined location for it.

Reply:
> There are various places where we can extract the underlying error, just want to make sure we have defined location for it.

Good point. I suspect this isn't the only place we're manipulating the underlying error. I suppose when we solve the relevant issues, we'll need to organize that. As far as the assumption you mentioned, that's true, but as long as the `UndefinedFunction` python class is always referring to the same undefined function error code in PostgreSQL, the first line of the message is the relevant one. It will always give the function call description, and tell you that the function doesn't exist.